### ISSUE TITLE
Feature request Ghostrider on GPU (OpenCL/Cuda)

### ISSUE THREAD

**Issue by phil2sat** [xmrig/xmrig#2744]
I know all the cons, but on my apu which is rated @25W i would be a nice benefit to use the APU for a little bit more hashes.

Actual everything seems fine, but: opencl   disabled (no suitable configuration found)

Thanks for your work

Phil2Sat

**Comment by Lonnegan** [comment]
If you use your APU's iGPU for mining additionally, however, that will slow down the CPU since they share a TDP budget. Whether that will be faster than just using the CPU?

**Comment by phil2sat** [comment]
We will only know if we prove it, maybe yes, mabe no.

**Comment by Spudz76** [comment]
Generally yes, most iGPU architectures share at least the memory bus, and sometimes a chunk of cache, which tend to clash.

If anyone ever makes an iGPU with on-die HBM entirely for it without sharing cache or memory with the CPU cores, then it would be quite viable.

**Comment by phil2sat** [comment]
since the ddr4 specs in xmrig infos and the actual hashrate on my 4700u, there should be plenty of bandwidth available

as far as i remember ddr4 3200 up to 6000H/s

4700u does average 1000H/s
guessing igpu does additional 100h/s

do the math..

And its not only for iGPU's some ppl want to mine on real cards

**Comment by nutsnox** [comment]
IMO This concept makes more sense for discrete GPU than iGPU as iGPU shares resources with the CPU and you could end up with worse performance.

For iGPU I would think that you'd have to pick one or the other to take the work (i.e. depending on what job you get and determining whether iGPU or CPU would be better). I would think that this logic would be really tough to mete out since iGPU performance varies across generations.

**Comment by Spudz76** [comment]
Pretty pointless splitting hairs, since supporting any GPU also supports iGPUs.  OpenCL is OpenCL...

**Comment by phil2sat** [comment]
Right, its not about implementing a new GPU, i want raptoreum protocol for gpu, since my card is already supported

**Comment by use-dashes-instead** [comment]
Spudz76 likes to hear himself talk, so don't worry too much about what he has to say

**Comment by phil2sat** [comment]
> Spudz76 likes to hear himself talk, so don't worry too much about what he has to say

pretty sure, yours was the only complete useless and off topic comment here. And what Spudz76 said is nothing more than a fact.

**Comment by use-dashes-instead** [comment]
> what Spudz76 said is nothing more than a fact.

It's not true, but carry on young trolls, carry on....

**Comment by phil2sat** [comment]
Mhmm, wonder why i can mine Ravencoin Kawpow on my VEGA iGPU at 1.91MH/s while my CPU does average 830H/s using XMRig.

Ah cause its not supported.

For the Speed, it has no Impact on Speed, running two instances, one for CPU and one for iGPU.

Under the line its dubling my crypto income for now.

The only thing i had to do was extra cooling for my laptop.

Request still open, Raptoreum config for GPU
